hack/go-tools-build.sh

```diff
@@ -37,7 +37,10 @@ if [ -z "${GOBIN}" ]; then
   exit 1
 fi
 
-CAPI_HACK_TOOLS="sigs.k8s.io/cluster-api/hack/tools"
+export GOWORK="off"
+
+GOMOD_REPLACE=${GOMOD_REPLACE:=}
```

Comment:
nit: A bit of godoc around these variables would be good

(in case we keep them, but I guess why not)